### PR TITLE
Agent discovery

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request implement the *Agent Discovery* functionality of the XRCE standard.
It is achieved using a UDP multicast server which can be added to the transport server as a module.

### CODE REVIEW

File: src/cpp/transport/udp/UDPServerLinux.cpp [hunk 1]
```diff
@@ -21,16 +21,18 @@
 #include <arpa/inet.h>
 #include <string.h>
 #include <errno.h>
+#include <ifaddrs.h>
```
Comment (BorjaOuterelo) [inline src/cpp/transport/udp/UDPServerLinux.cpp]: Is it needed?

File: src/cpp/types/XRCETypes.cpp [hunk 2]
```diff
@@ -5498,22 +5587,55 @@ size_t dds::xrce::ObjectInfo::getCdrSerializedSize(size_t current_alignment) con
...
-    dcdr >> m_activity;
+    bool m_config_flag;
+    dcdr >> m_config_flag;
+    if (m_config_flag)
+    {
+        ObjectVariant temp_config;
+        dcdr >> temp_config;
+        m_config = temp_config;
+    }
+    bool m_activity_flag;
```
Comment (BorjaOuterelo) [inline src/cpp/types/XRCETypes.cpp]: Could reuse the same bool from `m_config_flag`.

File: include/uxr/agent/transport/Server.hpp [hunk 3]
```diff
@@ -46,14 +46,17 @@ class Server
...
+    microxrcedds_agent_DllAPI bool stop();
 
     void push_output_packet(OutputPacket output_packet);
     virtual void on_create_client(EndPoint* source, const dds::xrce::ClientKey& client_key) = 0;
     virtual void on_delete_client(EndPoint* source) = 0;
     virtual const dds::xrce::ClientKey get_client_key(EndPoint* source) = 0;
     virtual std::unique_ptr<EndPoint> get_source(const dds::xrce::ClientKey& client_key) = 0;
 
+protected:
+    dds::xrce::TransportAddress transport_address_;
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply include/uxr/agent/transport/Server.hpp]: Done!

File: src/cpp/transport/udp/UDPServerLinux.cpp (hunk 1 above)
Comment (julianbermudez) [reply src/cpp/transport/udp/UDPServerLinux.cpp]: Removed!

File: src/cpp/types/XRCETypes.cpp (hunk 2 above)
Comment (julianbermudez) [reply src/cpp/types/XRCETypes.cpp]: Grouped!

**Pull request merged**